### PR TITLE
Nested packageProp path

### LINKED ISSUE #164: Nested packageProp path
Using `cosmiconfig` in a yarn version tool, [`yvm`](https://github.com/tophat/yvm). It would be helpful to be able to use the yarn version in the package.json engines prop as a search place for configuration. e.g.
```json
packageProp: 'engines.yarn',
```
A custom loader would probably work, but it seemed like a useful native feature regardless.

### PR DESCRIPTION
**Pull request by iamogbz**
- Allows targeting of nested `package.json` values using dot notation

Closes #164

### CODE REVIEW

File: src/createExplorer.js [hunk 1]
```diff
@@ -173,7 +174,7 @@ class Explorer {
 
   loadPackageProp(filepath: string, content: string) {
     const parsedContent = loaders.loadJson(filepath, content);
-    const packagePropValue = parsedContent[this.config.packageProp];
+    const packagePropValue = get(parsedContent, this.config.packageProp);
```
Comment (iamogbz) [inline src/createExplorer.js]: Not sure where to add the tests for the nested prop logic

Comment (davidtheclark) [reply src/createExplorer.js]: Tests will need to be added in all the places that `package.json` parsing is tested.

Comment (iamogbz) [reply src/createExplorer.js]: @davidtheclark done!

**Comment by iamogbz** [comment]
@davidtheclark is there anything else needed for this?

File: src/createExplorer.js (hunk 1 above)
Comment (olsonpm) [inline src/createExplorer.js]: We're taking a risk here that current devs aren't using a period in `packageProp`, which might be fine.  Just wanted to point out the possible breaking change.

**Comment by olsonpm** [comment]
@iamogbz - please update [the readme](https://github.com/davidtheclark/cosmiconfig/blob/master/README.md#packageprop) with a small blurb on allowing nested package properties

other than that it looks good to me.  At first I was weary of pulling in lodash.get because lodash utils are bigger than most people think - but its functionality is ubiquitous and thus expected, and the size isn't _that_ big.

**Comment by iamogbz** [comment]
> @iamogbz - please update [the readme](https://github.com/davidtheclark/cosmiconfig/blob/master/README.md#packageprop) with a small blurb on allowing nested package properties

@olsonpm updated, thanks!

File: src/createExplorer.js (hunk 1 above)
Comment (iamogbz) [reply src/createExplorer.js]: True, I guess we could use

```js
parsedContent[this.config.packageProp] || get(parsedContent, this.config.packageProp);
```

to mitigate the breaking change?

Comment (olsonpm) [reply src/createExplorer.js]: ha yeah I guess that's easy enough.  :thumbsup:

Comment (iamogbz) [reply src/createExplorer.js]: So it seems `lodash.get` already handles this case 😄 

Check it out https://runkit.com/iamogbz/5c60e0e12da4920012e32cad

Comment (olsonpm) [reply src/createExplorer.js]: oye - til I learned.  Thanks much for the quick code snippet.

**Comment by davidtheclark** [comment]
Thanks @iamogbz. It's looking good. I think there may be a bunch of additional tests worth writing for cases other than successfully loading a named file. Do you mind if I add a commit to this PR?

**Pull request merged**